pkg/proxy/mock_test.go

```diff
@@ -24,7 +24,7 @@ import (
 	"github.com/cilium/cilium/pkg/proxy/accesslog"
 )
 
-type proxyUpdaterMock struct {
+type ProxyUpdaterMock struct {
```

Comment:
Tried placing this in `pkg/testutils`, but unfortunately that introduced an import loop. Also, this will be removed from `pkg/proxy` by https://github.com/cilium/cilium/pull/12548.

Reply:
SGTM. I wonder if a new `pkg/testutils/proxy` package would make sense for this but if there's only going to be one instance soon anyway then the point is moot.